source/funkin/play/event/BopperSpeedEvent.hx

```diff
@@ -0,0 +1,85 @@
+package funkin.play.event;
+
+// Data from the chart
+import funkin.play.character.BaseCharacter;
+import funkin.play.stage.StageProp;
+import funkin.data.song.SongData;
+import funkin.data.song.SongData.SongEventData;
+// Data from the event schema
+import funkin.play.event.SongEvent;
+import funkin.data.event.SongEventSchema;
+import funkin.data.event.SongEventSchema.SongEventFieldType;
+
+import funkin.play.stage.Bopper;
+import flixel.FlxSprite;
+import flixel.util.FlxTimer;
+
+class BopperSpeedEvent extends SongEvent
+{
+  public function new()
+  {
+    super("BopperSpeed");
+  }
+
+  static final DEFAULT_DANCE_EVERY:Int = 1;
+
+  public override function handleEvent(data:SongEventData):Void
+  {
+    if (PlayState.instance == null || PlayState.instance.currentStage == null) return;
+
+    var bopperName:String = data.getString('bopper');
+
+    var speed:Int = data.getInt('danceEvery') ?? DEFAULT_DANCE_EVERY;
```

Comment:
We can just leave this as is for now, but in the private branch this has changed to a float which snaps to multiples of `0.25` rather than multiples of `1`.